core/trino-main/src/test/java/io/trino/operator/aggregation/TestApproximateSetVarchar.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.operator.aggregation;
+
+import io.airlift.slice.Slices;
+import io.trino.spi.type.Type;
+
+import java.util.concurrent.ThreadLocalRandom;
+
+import static io.trino.spi.type.VarcharType.VARCHAR;
+
+public class TestApproximateSetVarchar
+        extends AbstractTestApproximateSet
+{
+    @Override
+    protected Type getValueType()
+    {
+        return VARCHAR;
+    }
+
+    @Override
+    protected Object randomValue()
+    {
+        int length = ThreadLocalRandom.current().nextInt(100);
+        byte[] bytes = new byte[length];
+        ThreadLocalRandom.current().nextBytes(bytes);
+
+        return Slices.wrappedBuffer(bytes);
```

Comment:
It's unlikely to be a correct varchar value which must be valid utf-8

- you can call turn this class into varbinary test
- for varchar, you can choose codepoints (integer, not char) and then encode them into String with `java.lang.Character#toString(int)` and then into bytes with utf-8

Reply:
Stolen from `TestApproximateCountDistinctVarchar`. I will fix the former too.